docs/development/i18n.rst

```diff
@@ -72,19 +80,102 @@ the file can be found in the function that created it in
 ``securedrop/tests/pages-layout/test_journalist.py`` or
 ``securedrop/tests/pages-layout/test_source.py``.
 
-Merging translations
---------------------
+Merging translations back to develop
+------------------------------------
+
+`Weblate`_ automatically pushes the translations done via the web
+interface as a series of commit to the `i18n` branch in the `Weblate
+SecureDrop branch`_ which is a fork of the `develop` branch of the
+`SecureDrop git repository`_. These translations need to be submitted
+to the `develop` branch via pull requests for merge on a regular basis.
+
+.. code:: sh
+
+      $ git clone https://github.com/freedomofpress/securedrop
+      $ cd securedrop
+      $ git remote add lab http://lab.securedrop.club/bot/securedrop/tree/i18n
+      $ git fetch lab
+      $ git checkout -b wip-i18n origin/develop
+      $ h=$(git log --reverse --pretty='%H' --cherry-mark --right-only \
+            origin/develop...lab/i18n)
+      $ git cherry-pick $h
+      $ git push wip-i18n
```

Comment:
Thanks for writing this - this workflow seems reasonable! I have a concern regarding merging in translations after taking a look at #2366 - there will likely quickly become hundreds of automatic commits from Weblate littering our git history if we merge these automated commits as is. For example our last release was ~370 commits, which would quickly be swamped by automatic commits from translators doing their work. 

I took a look at how a few other projects, e.g. Tor and HTTPS everywhere, handle this issue to merge in translations, and I think they have a nice workflow - they have a [translation-only](https://gitweb.torproject.org/translation.git/) repository with one branch per project that commits are automatically pushed to when strings are translated. This repository is either added as a submodule, or changes are periodically [brought in](https://gitweb.torproject.org/bridgedb.git/commit/?id=fec3696d6826c8e8227e5286ffda5502249d3fc3) from the translation repo, bundling all the changes for one language at a time. 

Having a translation-only repository seems unnecessary given that you've set up `lab.securedrop.club` (:clap:), but squashing translation commits together into one commit per release per language for all the changes in that release is important in order for us to maintain the utility of our git history. This would add an additional step to the workflow above. What do you think?

Reply:
This sounds like an excellent idea ! I'll have to think a little more about the details but it is going to be more robust, won't clutter the git history and should not cause any merge problem because git should be able to cope with the commit asymmetry.

Reply:
I'm not so convinced this is the way to go anymore. The benefit of taking the time before making a decision ;-) I should point out that weblate does not create a commit each time a string is translated. That would be extremely noisy. When a translator starts work the language is locked. After a period of inactivity all the changes done during are committed into a single commit and pushed to the i18n branch. In other words we have a single commit per translation session.

At the beginning I made a lot of small changes and this may have given a different impression. I squashed all those into a single larger commit (https://github.com/freedomofpress/securedrop/pull/2366 now has ~100 commits instead of ~150). I stopped squashing when the first contributor started translating, to not remove his commit and preserve the attribution of his work.

These ~100 commits are spread over a period of ~100 days which is roughly one commit per day. I **hope** we can keep this level of engagement in the future ;-) And I also think is is unlikely to grow much more so we're in no real danger to drown under the localization commits.

# The proposed workflow

## Cons

* Localization commits are mixed with code commits and clutter the history

## Pros

* Translators are credited for their contribution in the main codebase
* a fork that is periodically merged back: there is no room for divergence (i.e a script that is in the translation repository but not in the securedrop repository)

# Workflow relying on squashed commits being copied from the translation repository to the secureDrop repository
 
## Cons

* merging/using translations with the latest from develop is more work than a merge
* translators contribution have to be acknowledged separately (i.e. not as code contributors)
* a separate repository is a risk of durable divergence

## Pros

* A single localization commit per release or so, less clutter

Reply:
Great writeup @dachary 🚀 . And thanks for the clarification on the commit per modification, that's good to know :+1:. My reservation still stands, currently we have a few languages with active translators/reviewers (~10 languages or so, thanks to your hard work), and as we get more activity, even one commit per translation session is going to be a a bit of noise (for comparison, Tor's Transifex has something like ~150 languages).

But I hear you, and I agree it's important that contributors are acknowledged, so I think we should add a note to the README linking to the contributions page (or equivalent) for the repository that get commits directly from Weblate (note: we could make a `securedrop-translations` repository in the `freedomofpress` group and have Weblate push commits to that if that is easier).

I _think_ divergence shouldn't be an issue if Weblate is watching `develop` in this repository and then either commits are cherry-picked from a fork and squashed or the relevant translation file is copied over, but let me know if I'm overlooking a way that divergence can be introduced. 

Summarizing, I propose we:

0. Squash together all commits for a particular language (or otherwise just copy the translations for a particular language since it's one file) and propose them as PRs for merge into the main repository to avoid clutter and follow the practice that e.g. [Signal](https://github.com/WhisperSystems/Signal-Android/commit/edc00d858b79e47ad8f9e658008455f621adafbc) and Tor are using to incorporate translations. 
1. Add a "How to Contribute Translations to SecureDrop" to the project `README.md` pointing to the translation portal and the contributors page for the fork that we use to get commits from Weblate so that people can see the translator and reviewer contribution history. 

Let me know your thoughts. And I'm happy to do this or help the doer with this! 😄

Reply:
Let's do it this way :-)I don't have strong feelings about this and what you're suggesting makes sense. Regarding contributions I propose we add an unsorted list of contributors who have at least one commit (code or translation) to the blog post announcing a release. This list is easy to compile for each repository. What do you think ?